File: solutions/3600-3699/3649.number-of-perfect-pairs/Solution.cpp
class Solution {
public:
    long long perfectPairs(vector<int>& nums) {
        int n = nums.size();

        vector<long long> arr;
        for(auto num : nums){
            long long x = num;
            arr.push_back(abs(x));
        }
        sort(arr.begin(), arr.end());

        long long count = 0;
        int j = 0;

        for(int i = 0; i < n; i++){
            while(j < n && arr[j] <= 2 * arr[i]){
                j++;
            }
            count += (j - i - 1);
        }
        return count;
    }
};

File: solutions/3600-3699/3649.number-of-perfect-pairs/Solution.java
class Solution {
    public long perfectPairs(int[] nums) {
        int n = nums.length;
        long[] arr = new long[n];
        for (int i = 0; i < n; i++) {
            arr[i] = Math.abs((long) nums[i]);
        }
        java.util.Arrays.sort(arr);

        long count = 0;
        int j = 0;
        for (int i = 0; i < n; i++) {
            while (j < n && arr[j] <= 2 * arr[i]) {
                j++;
            }
            count += (j - i - 1);
        }
        return count;
    }
}